libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/daemon/MessageBusListenerActor.scala

```diff
@@ -32,9 +36,11 @@ class MessageBusListenerActor[M](source: Source[M, NotUsed])(implicit messageLik
         log.warning("Listener already subscribed. Ignoring Subscribe message")
       case Failure(ex) =>
         log.error(ex, "Source/Listener died, subscribing again")
+        monitor.onError(ex)
         trySubscribeDelayed()
         context become idle
       case Done =>
+        monitor.onFinished
         log.info("Source finished, subscribing again")
         trySubscribeDelayed()
```

Comment:
Yeah that's true if it finishes with error we'll have a `Failure`, but `Future[Done]` is a normal stream state, the fact that kafka does not finish the stream does not mean we should just ignore that state. This is also useful for testing for example, where the stream actually finishes.